src/framebuffer/swtfb_client.rs

```diff
@@ -0,0 +1,224 @@
+//! This implements the IPC part of a rM2Framebuffer client to interact with this server:
+//! https://github.com/ddvk/remarkable2-framebuffer
+//!
+//! The client is developed according to the spec here:
+//! https://github.com/ddvk/remarkable2-framebuffer/issues/11
+
+use super::mxcfb::mxcfb_update_data;
+use crate::framebuffer::screeninfo::{FixScreeninfo, VarScreeninfo};
+use memmap2::{MmapOptions, MmapRaw};
+use std::ffi::{c_void, CString};
+use std::fs::{File, OpenOptions};
+use std::io::Error as IoError;
+use std::os::unix::prelude::AsRawFd;
+use std::{env, mem, ptr};
+
+const SWTFB_MESSAGE_QUEUE_ID: i32 = 0x2257c;
+
+pub const WIDTH: i32 = 1404;
+pub const HEIGHT: i32 = 1872;
+
+#[allow(non_upper_case_globals)]
```

Comment:
Why this instead of renaming or inlining?

Reply:
The globals are copied/ported from the [source implementation](https://github.com/ddvk/remarkable2-framebuffer/) (mainly `ipc.cpp` and `client/main.cpp`). So I opted to keep most things as close a possible rather than making the names more rust like.

Since many things in the lib also use C-like naming, I think it fits the lib more anyway.

Reply:
Tbh those could probably be cleaned up though, since the global is duplicated anyway. This is probably a rather extreme case of keeping the names. It's literally the same variable in two naming schemes right after another. Didn't even notice that. Good catch!

Reply:
Should be fixed with 88c97d7.